pkg/server/server_test.go

```diff
@@ -79,6 +82,45 @@ var _ = Describe("ApplicationsServer", func() {
 
 		Expect(res.Application.Name).To(Equal(name))
 	})
+	It("Authorize", func() {
+		ctx := context.Background()
+		provider := "github"
+		token := "token"
+
+		expectedToken, err := auth.Generate(auth.SecretKey, auth.ExpirationTime, provider, token)
+		Expect(err).NotTo(HaveOccurred())
+
+		res, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
+			ProviderName: provider,
+			AccessToken:  token,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(res.Token).To(Equal(expectedToken))
+	})
+	It("Authorize fails on wrong provider", func() {
+		ctx := context.Background()
+		provider := "wrong provider"
+		token := "token"
+
+		_, err := appsClient.Authenticate(ctx, &pb.AuthenticateRequest{
+			ProviderName: provider,
+			AccessToken:  token,
+		})
+		Expect(err.Error()).NotTo(Equal(fmt.Sprintf("unknown provider name %s, expecting github or gitlab", provider)))
```

Comment:
Seems like an odd assertion here. We are asserting that an error is NOT like this string? So it could be anything. Custom error types will make this safer.

Reply:
good catch 🙂